### PR TITLE
WIP: Notify via email and honeybadger if a fixity check fails.

### PR DESCRIPTION
**Pull request by eddierubeiz**
This is just waiting on:
    https://github.com/sciencehistory/scihist_digicoll/issues/394 . After that is taken care of it will be OK to merge / push this.

Ref #232

### CODE REVIEW

File: app/services/fixity_check_failure_service.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  def initialize(fixity_check)
+    raise ArgumentError.new("Fixity check passed was nil.") if fixity_check.nil?
+    raise ArgumentError.new("Please pass in a FixityCheck.") unless fixity_check.is_a? FixityCheck
+    @fixity_check = fixity_check
+    @asset = fixity_check.asset
+  end
+
+  def send
+    ActionMailer::Base.mail(from: "no-reply@sciencehistory.org",
+                            to: "digital-tech@sciencehistory.org",
```
Comment (jrochkind) [inline app/services/fixity_check_failure_service.rb]: Ideally email addressees woudln't be hard-coded in, but would be from local_env.yml somewhere. Not sure if we already have keys for these though, not that big a deal to hard-code them in.

**Comment by jrochkind** [comment]
I'm not totally sure why I didn't in chf_sufia do things the 'normal' way with an ActionMailer sub-class and and template file. Instead putting the 'template' inline in a string. 

Maybe cause I thought I was gonna reuse the plain text template for more than one delivery, not just email? Or cause it seemed so simple there was no need to put it in a separate template file?

Here it is non-trivial and is only used for email... it kind of seems like it should be a normal ActionMailer sub-class. but I dunno, maybe it's good enough like this, not worth fixing? Could run into unexpected problems doing it the 'normal' way, and it is working now, so. What do you think?

**Comment by eddierubeiz** [comment]
I like the idea of standardizing this so it's more in line with Rails best practices. But I'd probably vote for making that refactor a GitHub issue and putting that in our post-launch to-do list.

**Comment by eddierubeiz** [comment]
I put the refactor in a separate ticket so we can merge this PR after I fix the hard-coded emails.

File: app/services/fixity_check_failure_service.rb (hunk 1 above)
Comment (eddierubeiz) [reply app/services/fixity_check_failure_service.rb]: Fixed!

**Comment by jrochkind** [comment]
Eddie writes in Slack:

> Eddie Rubeiz 3:00 PM
> Added the email addresses for fixity check email failures to the staging branch version of the local_env.yml template in Ansible

> https://bitbucket.org/ChemicalHeritageFoundation/ansible-inventory/commits/8034aa85e7cdaa17eea7a8da7cd88a56f1b458e7/raw

That seems to be a link to a specific commit, not a PR, in bitbucket. Not sure if bitbucket uses PR's or what. 

Is it okay to merge this PR #387 before that ansible change? Or will deploying this code without the ansible change result in errors?

I am also not sure where we are with the situation where emails do not actually arrive. @eddierubeiz  could you update on that here?  We can still merge the code even if emails never arrive, but we can't close the issue without either getting stakeholder sign-off on email alerts not working, or making them work. Can you keep track of that and create necessary issues and such to track and bring them up with stakeholders, @eddierubeiz ?

**Comment by eddierubeiz** [comment]
Good call! Updated the ticket with the two preconditions for merge above. Basically the jobs server should be able to send emails, and the to addresses (no longer hardcoded) need to be added by Ansible to the `local_env.rb` settings file.

**Comment by jrochkind** [comment]
@eddierubeiz will move error emails to a bg job (ie use `deliver_later`), and then we can safely deploy this even though email sending currently raises an error. (Pending #394)

**Comment by eddierubeiz** [comment]
Now using `deliver_later`; @jrochkind, you may merge at your convenience.

**Pull request merged**